### PR TITLE
Refactor QR scanner to support WalletConnect integration

### PR DESCRIPTION
**Pull request by milsawicki**
## Purpose
In order to connect _WalletConnect_ you have to scan QR code. We already had a class that was responsive for scanning QR wallet addresses but was not expandable/reusable. The class has been refatored to support _WalletConnect._   
## Changes
- Add QR scan button in _accounts_ view which is supposed to scan WalletConnect codes.
- Removed separate ScanQR views from `SendFunds.storyboard` and `Account.storyboard`
- Refactored ScanAddressQRViewController to ScanQRViewController to make the view reusable by injecting stratedy 
## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: ConcordiumWallet/AppCoordinator.swift [hunk 1]
```diff
@@ -255,6 +254,20 @@ class AppCoordinator: NSObject, Coordinator, ShowAlert, RequestPasswordDelegate
...
+    func showWalletConnectScanner() {
+        let vc = ScanQRViewControllerFactory.create(
+            with: ScanQRPresenter(
+                strategy: WalletConnectStrategy(),
+                didScanQrCode: { [weak self] address in
+                    // Successfully scanner WalletConnect QR.
+                    // TODO: Handle Wallet Connect logic here
+                    self?.navigationController.popViewController(animated: true)
+                }
+            )
```
Comment (milsawicki) [inline ConcordiumWallet/AppCoordinator.swift:266]: Will be handled in further PRs

File: ConcordiumWallet/Service/MobileWallet.swift [hunk 2]
```diff
@@ -131,9 +131,7 @@ class MobileWallet: MobileWalletProtocol {
             let revealedAttributes = account.revealedAttributes.map({
                 $0.key
             })
-    
-//            return requestPasswordDelegate.requestUserPassword(keychain: keychain)
-//            .flatMap { (pwHash) -> AnyPublisher<CreateCredentialRequest, Error> in
+
```
Comment (milsawicki) [inline ConcordiumWallet/Service/MobileWallet.swift:134]: Just a clean up of unused leftsovers trash code...

File: ConcordiumWallet/Storyboards/Base.lproj/Account.storyboard [hunk 3]
```diff
@@ -1,9 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (milsawicki) [inline ConcordiumWallet/Storyboards/Base.lproj/Account.storyboard:1]: these changes derive from removing a AddressQRScannerViewController blue print from storyboard.
Now we use QR scanner view entierly programatically.

File: ConcordiumWallet/Storyboards/SendFund.storyboard [hunk 4]
```diff
@@ -1,9 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<document type="com.apple.InterfaceBuilder3.CocoaTouch.Storyboard.XIB" version="3.0" toolsVersion="20037" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" useSafeAreas="YES" colorMatched="YES">
```
Comment (milsawicki) [inline ConcordiumWallet/Storyboards/SendFund.storyboard:2]: same as above

File: ConcordiumWallet/Views/AccountsView/AccountDetails/AccountDetailsViewController.swift [hunk 5]
```diff
@@ -347,11 +347,6 @@ class AccountDetailsViewController: BaseViewController, AccountDetailsViewProtoc
 
 extension AccountDetailsViewController {
     func setupTabBar() {
-//        tabViewModel.tabs = [
-//            "accountDetails.transfers".localized,
-//            "accountDetails.identity_data".localized
-//        ]
-
```
Comment (milsawicki) [inline ConcordiumWallet/Views/AccountsView/AccountDetails/AccountDetailsViewController.swift:354]: cleanup ...

File: ConcordiumWallet/Views/MoreSection/MoreCoordinator.swift [hunk 6]
```diff
@@ -208,28 +199,24 @@ extension MoreCoordinator: AddRecipientPresenterDelegate {
...
 
-extension MoreCoordinator: RequestPasswordDelegate {
-}
+extension MoreCoordinator: RequestPasswordDelegate {}
 
 extension MoreCoordinator: InitialAccountInfoPresenterDelegate {
     func userTappedClose() {
         navigationController.popToRootViewController(animated: true)
     }
-
```
Comment (milsawicki) [inline ConcordiumWallet/Views/MoreSection/MoreCoordinator.swift:232]: formatting, just ignore :)

File: ConcordiumWallet/Views/MoreSection/MoreCoordinator.swift [hunk 7]
```diff
@@ -84,49 +90,30 @@ class MoreCoordinator: Coordinator, ShowAlert, MoreCoordinatorDelegate {
...
-//        let vc = ExportFactory.create(with: ExportPresenter(dependencyProvider: dependencyProvider, requestPasswordDelegate: self, delegate: self))
-//        navigationController.pushViewController(vc, animated: true)
-//    }
-    
-//    func showValidateIdsAndAccounts() {
-//        sanityChecker.requestPwAndCheckSanity(requestPasswordDelegate: self,
-//                                              keychainWrapper: dependencyProvider.keychainWrapper(),
-//                                              mode: .manual)
-//    }
-
```
Comment (milsawicki) [inline ConcordiumWallet/Views/MoreSection/MoreCoordinator.swift:107]: leftovers...

File: ConcordiumWallet/Views/AccountsView/AccountsViewController.swift [hunk 8]
```diff
@@ -178,52 +183,52 @@ class AccountsViewController: BaseViewController, Storyboarded, AccountsViewProt
...
+
         viewModel.$totalBalance
-                .map { $0.displayValueWithGStroke() }
-                .assign(to: \.text, on: totalBalanceLabel)
-                .store(in: &cancellables)
-        
+            .map { $0.displayValueWithGStroke() }
+            .assign(to: \.text, on: totalBalanceLabel)
+            .store(in: &cancellables)
+
```
Comment (milsawicki) [inline ConcordiumWallet/Views/AccountsView/AccountsViewController.swift:206]: formatting

File: ConcordiumWallet/Views/AccountsView/AccountsPresenter.swift [hunk 9]
```diff
@@ -220,7 +226,11 @@ class AccountsPresenter: AccountsPresenterProtocol {
     func showSettings() {
         delegate?.showSettings()
     }
-
+    
+    func showWalletConnectScanner() {
+        delegate?.showWalletConnectScanner()
```
Comment (milsawicki) [inline ConcordiumWallet/Views/AccountsView/AccountsPresenter.swift:231]: think about callilng on AccountsCoordinator and not the AppCoordinator

**Review by bisgardo — APPROVED** [review]

File: ConcordiumWallet/Views/ScanQR/Strategies/WalletConnectStrategy.swift [hunk 10]
```diff
@@ -0,0 +1,23 @@
+//
+//  WalletConnectStrategy.swift
+//  Mock
+//
+//  Created by Milan Sawicki on 31/05/2023.
+//  Copyright © 2023 concordium. All rights reserved.
+//
+
+class WalletConnectStrategy: QRScannerStrategy {
```
[the remark this answers is not in the document]
Comment (bisgardo) [reply ConcordiumWallet/Views/ScanQR/Strategies/WalletConnectStrategy.swift]: this has been refactored

File: ConcordiumWallet/Views/ScanQR/ScanQRPresenter.swift [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+    var view: ScanQRViewProtocol? { get set }
+    func viewDidLoad()
+    func scannedQrCode(_: String)
+}
+
+class ScanQRPresenter: ScanQRPresenterProtocol {
+    weak var view: ScanQRViewProtocol?
+    weak var delegate: QRCodeStrategyDelegate?
+    var strategy: QRScannerStrategy
+    var didScanQrCode: (_ address: String) -> Void
```
[the remark this answers is not in the document]
Comment (bisgardo) [reply ConcordiumWallet/Views/ScanQR/ScanQRPresenter.swift]: fixed

File: ConcordiumWallet/Views/ScanQR/ScanQRViewController.swift [hunk 12]
```diff
@@ -129,17 +81,73 @@ extension ScanAddressQRViewController: AVCaptureMetadataOutputObjectsDelegate {
...
         scanGuide.tintColor = .green
     }
 
     func showQrInvalid() {
         scanGuide.tintColor = .red
         showToast(withMessage: "scanQr.invalidQr".localized)
-        self.captureSession.startRunning()
+        DispatchQueue.global(qos: .background).async { [weak self] in
+            self?.captureSession.startRunning()
+        }
```
[the remark this answers is not in the document]
Comment (milsawicki) [reply ConcordiumWallet/Views/ScanQR/ScanQRViewController.swift:93]: The camera capture session should be run from background thread in order to not block main thread, so it doesn't hang/lag the UI when the view appears :)

**Pull request merged**